File: Python Pattern Programs/Wave Patterns/Pattern 4.py
waveHeight = 4  # 1
wL = 4  # 2
wH = waveHeight - 1  # 3
ctrl = wH  # 4

for x in range(0, wH + 1):
    cp = ord('A')  # 5
    for y in range(0, (wH * wL * 2) + 1):
        if (y % (wH * 2) == ctrl or
                y % (wH * 2) == wH + x):
            print(chr(cp), end="")
        else:
            print(" ", end="")  # single space
        cp += 1  # 6
        if (cp > ord('Z')):
            cp = cp - 26  # 7
    ctrl -= 1
    print()

    """
1) change value to increase/decrease the height of the wave
2) change value to increase/decrease the length of the wave
3) for loop cond
4) cond for printing
5) set printing character
6) increment print char
7) reset print char to 'A'
    """

File: JavaPatternPrograms/WavePatterns/Pattern 4.java
class Wave_4
{
   public static void main(String args[])
   {
    int waveHeight = 4; //change value to increase or decrease the height of wave


    int wL = 4; //wave length->change value to increase or decrease the length of wave


    int wH = waveHeight - 1; //for loop cond.

    int x = wH; //if cond for printing
    int cp; //print char


    for (int i = 0;i <= wH;i++)
    {
     cp = 'A'; // set print char.

     for (int j = 0;j <= wH * wL * 2;j++)
     {
        if (j % (wH * 2) == x || j % (wH * 2) == wH + i)
        {
         System.out.print((char)cp);
        }
         else
         {
         System.out.print(" ");
         }

      cp++; // increment print char

      /reset print char to 'A'/
      if (cp > 'Z')
      {
         cp = cp - 26;
      }

     }
     x--;
     System.out.println();
    }
    
   }
}